### PR TITLE
Add section on Python debugging and clangd to VS Code docs

### PR DESCRIPTION
**Pull request by DavidFair**
Adds sections on attaching to Python code currently executing and using clangd for live warnings and errors

**To test:**
Proof read, does it make sense

Fixes N/A

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
Just a couple of missed words below. 

As discussed in person, there are a few more steps to this that need adding. Here's the CMake command: 
`CXX=clang++ CC=clang cmake -S <src_dir> -B <build_dir> -DPYTHON_EXECUTATBLE=/usr/bin/python2/3`

**Comment by hsautessella** [comment]
Since today's my last day working on Mantid, I'm unassigning myself

File: dev-docs/source/VSCode.rst [hunk 1]
```diff
@@ -355,3 +420,41 @@ Reference" and hit Enter.
...
 +-------------------+---------------+---------------+---------------+
+
+(Advanced) Getting Live Warnings and Errors with Clangd
+=======================================================
+(Linux only)
+
+The C++ extension in VS Code provides limited inspection: it (currently) has
+warnings disabled and will only emit errors.
+
+Clang be used to provide live warnings and will notify on common bugs, like
```
Comment (DTasev) [inline dev-docs/source/VSCode.rst]:
```suggestion
Clang can be used to provide live warnings and will notify on common bugs, like
```

File: dev-docs/source/VSCode.rst [hunk 2]
```diff
@@ -355,3 +420,41 @@ Reference" and hit Enter.
...
+implicit casts, which are normally only picked building.
```
Comment (DTasev) [inline dev-docs/source/VSCode.rst]:
```suggestion
implicit casts, which are normally only picked up upon building.
```

**Review by dtasev — DISMISSED** [review]
Just made Harry's comments into code suggestions to hopefully speed this PR up.

**Review by NickDraper — APPROVED** [review]
This looks good an incorporates the changes suggested by harry

**Pull request merged**